activestorage/app/models/active_storage/current.rb

```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 class ActiveStorage::Current < ActiveSupport::CurrentAttributes #:nodoc:
-  attribute :host
+  attribute :url_options
```

Comment:
I think what @georgeclaghorn was indicating is that we also do the parsing in this class, like this:

```suggestion
  attribute :url_options

  def set_url_options_from(host)
    self.url_options = URI.parse(host).then do |uri|
      { protocol: uri.scheme, host: uri.host, port: uri.port }
    end
  end
end
```

Reply:
I don’t think we need to do any parsing. We can set the right URL options based on `request` in `SetCurrent`.

Reply:
Ah lol, of course 👍